src/mol-model-formats/structure/util.ts

```diff
@@ -47,30 +47,31 @@ export function guessElementSymbolTokens(tokens: Tokens, str: string, start: num
 }
 
 const reTrimSpacesAndNumbers = /^[\s\d]+|[\s\d]+$/g;
-export function guessElementSymbolString(str: string) {
+export function guessElementSymbolString(atomId: string, compId: string) {
     // trim spaces and numbers, convert to upper case
-    str = str.replace(reTrimSpacesAndNumbers, '').toUpperCase();
-    const l = str.length;
+    atomId = atomId.replace(reTrimSpacesAndNumbers, '').toUpperCase();
+    const l = atomId.length;
 
-    if (l === 0) return str; // empty
-    if (l === 1) return str; // one char
+    if (l === 0) return atomId; // empty
+    if (l === 1) return atomId; // one char
 
     if (l === 2) { // two chars
-        if (str === 'NA' || str === 'CL' || str === 'FE' || str === 'SI' ||
-            str === 'BR' || str === 'AS'
-        ) return str;
+        if (atomId === 'NA' || atomId === 'CL' || atomId === 'FE' || atomId === 'SI' ||
+            atomId === 'BR' || atomId === 'AS'
+        ) return atomId;
     }
 
-    if (l === 3) { // three chars
-        if (str === 'SOD') return 'NA';
-        if (str === 'POT') return 'K';
-        if (str === 'CES') return 'CS';
-        if (str === 'CAL') return 'CA';
-        if (str === 'CLA') return 'CL';
+    if (l === 3 && compId === atomId) { // three chars
```

Comment:
Where does the ``compId === atomId`` condition come from? Seems a bit weird to me...

Reply:
Yeah, that warrants a comment. It is to correctly assign elements for ions from files with Charmm FF names. For those ions compId and atomId are the same.